packages/SwingSet/src/kernel/virtualObjectManager.js

```diff
@@ -364,8 +364,9 @@ export function makeVirtualObjectManager(
           // exported non-virtual object: Remotable
           const remotable = requiredValForSlot(vref);
           if (remotableRefCounts.has(remotable)) {
-            /** @type {number} */
-            const oldRefCount = (remotableRefCounts.get(remotable));
+            const oldRefCount = /** @type {number} */ (remotableRefCounts.get(
```

Comment:
Why did this cast need to move?

I'd suggest refactoring so no casting is necessary,
```
          const oldRefCount = remotableRefCounts.get(remotable) || 0;
          remotableRefCounts.set(remotable, oldRefCount);
```

Reply:
It wasn't a cast before. It was trying to assign a `number | undefined` to a `number` so it blew up. Not sure why it even worked in the first place.

Reply:
@FUDCo @warner, happy to adopt @turadg's suggestion but it's technically a behavioral change I wanted to avoid.

Reply:
If it's just to ensure that it's not undefined then I think we'd be better off with [NonNullish](https://github.com/Agoric/agoric-sdk/pull/4422) so we'd have a runtime guarantee

Reply:
since non-nullish hasn't landed yet, I added a manual assertion to second cast where I believe it was warranted.